### PR TITLE
Implement session timeout after 60 mins inactivity

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-840

### CODE REVIEW

File: app/controllers/hiring_staff/base_controller.rb [hunk 1]
```diff
@@ -28,4 +32,23 @@ def current_user
 
     @current_user ||= User.find_or_create_by(oid: current_session_id)
   end
+
+  def check_user_last_activity_at
+    return redirect_to dsi_logout_url if current_user&.last_activity_at.blank?
+
+    if Time.zone.now > (current_user.last_activity_at + TIMEOUT_PERIOD)
+      session[:signing_out_for_inactivity] = true
+      redirect_to dsi_logout_url
```
Comment (david-mears-dfe) [inline app/controllers/hiring_staff/base_controller.rb:41]: I think perhaps this should be `return redirect_to`, in order to make the action `check_user_last_activity_at` closed and thus avoid the possibility of too many redirects.

File: app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb [hunk 2]
```diff
@@ -18,8 +19,13 @@ def create
...
 
   def destroy
+    if session[:signing_out_for_inactivity]
+      flash_message = { notice: I18n.t('messages.access.signed_out_for_inactivity') }
+    else
+      flash_message = { success: I18n.t('messages.access.signed_out') }
+    end
     session.destroy
-    redirect_to root_path, notice: I18n.t('messages.access.signed_out')
+    redirect_to new_identifications_path, flash_message
```
Comment (david-mears-dfe) [inline app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb:28]: I think perhaps this should be `return redirect_to`, in order to make the action `destroy` closed and thus avoid the possibility of too many redirects.

Comment (cesidio) [reply app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb:28]: No need to return on the last line of the method

File: app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb [hunk 3]
```diff
@@ -18,8 +19,13 @@ def create
   end
 
   def destroy
+    if session[:signing_out_for_inactivity]
+      flash_message = { notice: I18n.t('messages.access.signed_out_for_inactivity') }
+    else
+      flash_message = { success: I18n.t('messages.access.signed_out') }
```
Comment (cpjmcquillan) [inline app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb:25]: Check with UX, but I think this should be `notice` for both

File: config/locales/en.yml [hunk 4]
```diff
@@ -648,6 +648,7 @@ en:
         only_published: You can only view published jobs
     access:
       signed_out: You've been signed out.
+      signed_out_for_inactivity: You've been signed out due to inactivity.
```
Comment (cpjmcquillan) [inline config/locales/en.yml:651]: You have been and no full stops

File: app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb (hunk 3 above)
Comment (david-mears-dfe) [reply app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb]: Correct https://ukgovernmentdfe.slack.com/archives/CG9A8H1HP/p1589793769222100?thread_ts=1589793583.221500&cid=CG9A8H1HP

**Pull request merged**